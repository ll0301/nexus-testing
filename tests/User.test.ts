import { createTestContext } from './__helpers'

const ctx = createTestContext()

it('mutation - signUpUser', async () => {
  const signUpUserResult = await ctx.client.send(`
    mutation {
      signUpUser(email: "jenny@email.com", password: "123456") {
        id
        email
        password
        isSignIn
      }
    }
  `)
  
  expect(signUpUserResult).toMatchInlineSnapshot(`
    Object {
      "signUpUser": Object {
        "email": "jenny@email.com",
        "id": "cuid4",
        "isSignIn": false,
        "password": "123456",
      },
    }
  `)
})

it('query - users', async () => {
  const usersResult = await ctx.client.send(`
    query {
      users {
        id
        email
        isSignIn
        password
      }
    }
  `)

  expect(usersResult).toMatchSnapshot();
})

it('mutation - signInUser', async () => {
  const signInUserResult = await ctx.client.send(`
    mutation {
      signInUser(email: "jenny@email.com", password: "123456") {
        id
        email
        password
        isSignIn
      }
    }
  `)
  
  expect(signInUserResult).toMatchInlineSnapshot(`
    Object {
      "signInUser": Object {
        "email": "jenny@email.com",
        "id": "cuid4",
        "isSignIn": true,
        "password": "123456",
      },
    }
  `)
})

it('query - signInUsers', async () => {
  const signInUsersResult = await ctx.client.send(`
    query {
      signInUsers {
        email
        id
        isSignIn
        password
      }
    }
  `)

  expect(signInUsersResult).toMatchSnapshot();
  expect(signInUsersResult.signInUsers[0].isSignIn).toBeTruthy();
})