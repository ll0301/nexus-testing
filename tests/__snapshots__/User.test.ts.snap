// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query - signInUsers 1`] = `
Object {
  "signInUsers": Array [
    Object {
      "email": "jenny@email.com",
      "id": "cuid4",
      "isSignIn": true,
      "password": "123456",
    },
  ],
}
`;

exports[`query - users 1`] = `
Object {
  "users": Array [
    Object {
      "email": "jerry@email.com",
      "id": "cuid1",
      "isSignIn": false,
      "password": "123456",
    },
    Object {
      "email": "dean@email.com",
      "id": "cuid2",
      "isSignIn": false,
      "password": "123456",
    },
    Object {
      "email": "daniel@email.com",
      "id": "cuid3",
      "isSignIn": false,
      "password": "123456",
    },
    Object {
      "email": "jenny@email.com",
      "id": "cuid4",
      "isSignIn": false,
      "password": "123456",
    },
  ],
}
`;
